### PR TITLE
Simplify creation of unary structured kernels.

### PR DESCRIPTION
**Pull request by hameerabbasi**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#54252 Simplify creation of unary structured kernels.**

Differential Revision: [D27265348](https://our.internmc.facebook.com/intern/diff/D27265348)

### CODE REVIEW

File: aten/src/ATen/native/UnaryOps.cpp [hunk 1]
```diff
@@ -26,11 +26,13 @@ namespace at {
 
 namespace meta {
 
-TORCH_META_FUNC(sin) (
-  const Tensor& self
-) {
-  build_unary_float_op(maybe_get_output(), self);
-}
+#define CREATE_UNARY_META_FUNC(func)                  \
```
Comment (bdhirsh) [inline aten/src/ATen/native/UnaryOps.cpp:29]: This is probably fine if most unary ops follow this convention, but any unary ops that, e.g., perform their own error checking shouldn't use this.

**Review by bdhirsh — APPROVED** [review]

File: aten/src/ATen/native/UnaryOps.cpp (hunk 1 above)
Comment (hameerabbasi) [reply aten/src/ATen/native/UnaryOps.cpp:29]: Would you like me to add a comment?

Comment (bdhirsh) [reply aten/src/ATen/native/UnaryOps.cpp:29]: yeah, that couldn't hurt :)

Comment (bdhirsh) [reply aten/src/ATen/native/UnaryOps.cpp:29]: You might also have to rebase- I'm seeing merge conflicts in internal tests on `native_functions.yaml`

**Comment by hameerabbasi** [comment]
@bdhirsh Superceded by #54592, `ghstack` created a new PR for some reason.

**Pull request closed without merging**